linkerd/http-route/src/grpc/tests.rs

```diff
@@ -0,0 +1,171 @@
+use super::{r#match::*, *};
+use crate::http::MatchHeader;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum Policy {
+    Expected,
+    Unexpected,
+}
+
+impl Default for Policy {
+    fn default() -> Self {
+        Self::Unexpected
+    }
+}
+
+/// Given two equivalent routes, choose the explicit hostname match and not
+/// the wildcard.
+#[test]
+fn hostname_precedence() {
+    let rts = vec![
+        Route {
+            hosts: vec!["*.example.com".parse().unwrap()],
+            rules: vec![Rule {
+                matches: vec![MatchRoute {
+                    rpc: MatchRpc {
+                        service: Some("foo".to_string()),
+                        method: Some("bar".to_string()),
+                    },
+                    ..MatchRoute::default()
+                }],
+                ..Rule::default()
+            }],
+        },
+        Route {
+            hosts: vec!["foo.example.com".parse().unwrap()],
+            rules: vec![Rule {
+                matches: vec![MatchRoute {
+                    rpc: MatchRpc {
+                        service: Some("foo".to_string()),
+                        method: Some("bar".to_string()),
+                    },
+                    ..MatchRoute::default()
+                }],
+                policy: Policy::Expected,
+            }],
+        },
+    ];
+
+    let req = http::Request::builder()
+        .method(http::Method::POST)
+        .uri("http://foo.example.com/foo/bar")
+        .body(())
+        .unwrap();
+    let (_, policy) = find(&rts, &req).expect("must match");
+    assert_eq!(*policy, Policy::Expected, "incorrect rule matched");
+}
+
+#[test]
+fn method_precedence() {
+    // Given two equivalent routes, choose the longer path match.
```

Comment:
very silly nit: most of the tests in this file have doc comments outside of the test, but this one has a non-doc-comment inside of the test. let's change this to
```suggestion
/// Given two equivalent routes, choose the longer path match.
#[test]
fn method_precedence() {
```
for consistency?